src/libstore/build.cc

```diff
@@ -4526,8 +4545,15 @@ void SubstitutionGoal::tryToRun()
             Activity act(*logger, actSubstitute, Logger::Fields{worker.store.printStorePath(storePath), sub->getUri()});
             PushActivity pact(act.id);
 
+            auto subPath = storePath;
+            if (ca) {
+                subPath = sub->makeFixedOutputPathFromCA(storePath.name(), *ca);
```

Comment:
Since we already computed `subPath` in `SubstitutionGoal::tryNext()`, we could avoid computing it again by storing `subPath` in `SubstitutionGoal`.